src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java

```diff
@@ -0,0 +1,44 @@
+package com.wikia.webdriver.elements.oasis.components.comment;
+
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
+import com.gargoylesoftware.htmlunit.ElementNotFoundException;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class ArticleComment extends BasePageObject {
+
+  @FindBy(css = "#article-comm-submit")
+  protected WebElement commentSubmitButton;
+  String videoInCommentsSelector = ".speech-bubble-message img[data-video-name*='%videoName%']";
+
+
+  public ArticleComment waitForVideo() {
+    driver.switchTo().frame(1);
+    wait.forElementVisible(By.cssSelector("img"));
+
+    return this;
+  }
+
+  public ArticleComment submitComment() {
+    driver.switchTo().defaultContent();
+    scrollAndClick(commentSubmitButton);
+    waitForElementNotVisibleByElement(commentSubmitButton, 30);
+    PageObjectLogging.log("submitComment", "comment has been submitted", true);
+
+    return this;
+  }
+
+  public boolean isVideoVisible(String videoName) {
+    try {
+      WebElement
+          element =
+          driver.findElement(
+              By.cssSelector(videoInCommentsSelector.replace("%videoName%", videoName)));
+      return element.isDisplayed();
+    } catch (ElementNotFoundException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)